### DISCUSSION TITLE
Restart docker-compose image

### DISCUSSION THREAD

**Discussion by iloveitaly** [community/community#34090] [Codespaces]
Is there anyway to run the equivalent of:

```
docker compose up -d
```

Against the `docker-compose.yml` contained within the `.devcontainer/`? Or is the only way to rebuild the entire container?

**Comment by salvador-barboza** [comment]
Hello! 

If you are using the `docker-compose.yml` file in your `devcontainer.json`'s `dockerComposeFile`, then you will need to rebuild your Codespace when you make any changes.


[`devcontainer.json` docs](https://containers.dev/implementors/json_reference/#compose-specific)

**Comment by iloveitaly** [comment]
It looks like the only real solution here is to run compose from within your container as described here https://notes.alexkehayias.com/running-docker-compose-in-codespaces/
